python/ray/serve/http_adapters.py

```diff
@@ -0,0 +1,59 @@
+from io import BytesIO
+from typing import List, Optional, Union
+
+from fastapi import File
+from pydantic import BaseModel, Field
+
+from ray.serve.model_wrappers import DataBatchType
+from ray.serve.utils import require_packages
+
+
+_1DArray = List[float]
+_2DArray = List[List[float]]
+_3DArray = List[List[List[float]]]
+_4DArray = List[List[List[List[float]]]]
+
+
+class NdArray(BaseModel):
+    """Schema for numeric array input."""
+
+    array: Union[_1DArray, _2DArray, _3DArray, _4DArray] = Field(
+        ...,
+        description=(
+            "The array content as a list or nested list. "
+            "You can pass in 1D to 4D array in its original form or the flatten verion."
+        ),
+    )
+    shape: Optional[List[int]] = Field(
+        default=None,
+        description=("The shape of the array. If present, the array will be reshaped."),
+    )
+    dtype: Optional[str] = Field(
```

Comment:
what are the available values?

Reply:
can we make it an enum?

Reply:
tried. there are like 158 of them. User will get a nice validation error from numpy anyway and with 158 types on documentation won't be very useful.
```
>>> import numpy as np
>>> np.sctypeDict.keys()
dict_keys(['?', 0, 'byte', 'b', 1, 'ubyte', 'B', 2, 'short', 'h', 3, 'ushort', 'H', 4, 'i', 5, 'uint', 'I', 6, 'intp', 'p', 7, 'uintp', 'P', 8, 'long', 'l', 'L', 'longlong', 'q', 9, 'ulonglong', 'Q', 10, 'half', 'e', 23, 'f', 11, 'double', 'd', 12, 'longdouble', 'g', 13, 'cfloat', 'F', 14, 'cdouble', 'D', 15, 'clongdouble', 'G', 16, 'O', 17, 'S', 18, 'unicode', 'U', 19, 'void', 'V', 20, 'M', 21, 'm', 22, 'bool8', 'Bool', 'b1', 'int64', 'Int64', 'i8', 'uint64', 'Uint64', 'u8', 'float16', 'Float16', 'f2', 'float32', 'Float32', 'f4', 'float64', 'Float64', 'f8', 'float128', 'Float128', 'f16', 'complex64', 'Complex32', 'c8', 'complex128', 'Complex64', 'c16', 'complex256', 'Complex128', 'c32', 'object0', 'Object0', 'bytes0', 'Bytes0', 'str0', 'Str0', 'void0', 'Void0', 'datetime64', 'Datetime64', 'M8', 'timedelta64', 'Timedelta64', 'm8', 'UInt64', 'int32', 'Int32', 'i4', 'uint32', 'UInt32', 'u4', 'int16', 'Int16', 'i2', 'uint16', 'UInt16', 'u2', 'int8', 'Int8', 'i1', 'uint8', 'UInt8', 'u1', 'complex_', 'int0', 'uint0', 'single', 'csingle', 'singlecomplex', 'float_', 'intc', 'uintc', 'int_', 'longfloat', 'clongfloat', 'longcomplex', 'bool_', 'bytes_', 'string_', 'str_', 'unicode_', 'object_', 'int', 'float', 'complex', 'bool', 'object', 'str', 'bytes', 'a'])
```

Reply:
I see. Can you update the description to specify that this is using the numpy dypte?